partiql-ast/src/main/java/org/partiql/ast/v1/expr/ExprDateDiff.java

```diff
@@ -0,0 +1,44 @@
+package org.partiql.ast.v1.expr;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.v1.AstNode;
+import org.partiql.ast.v1.AstVisitor;
+import org.partiql.ast.v1.DatetimeField;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * TODO docs, equals, hashcode
+ */
+public class ExprDateDiff extends Expr {
```

Comment:
Really hope we don't *need* to add this to the AST. Same goes for `DATE_ADD`. I wonder if there's a way we can rewrite this to be `+/-` for datetimes and intervals.

Reply:
From offline discussion: we'll remove `DATE_ADD` and `DATE_DIFF` from the AST with the eventual plan to support the `+/-` syntax with intervals.

To support customers dependent on these existing functions and special keyword syntax, we will still support parsing the special syntax but
- convert the datetime keyword (e.g. `month`, `day`) into a string or
- map to different functions for each variant (e.g. `date_add_month`, `date_diff_day`)

`DATE_ADD` and `DATE_DIFF` will still have support in the plan and evaluator.